src/poll.rs

```diff
@@ -225,7 +226,7 @@ use std::{fmt, io};
 /// [`SourceFd`]: unix/struct.SourceFd.html
 /// [`Poll::poll`]: struct.Poll.html#method.poll
 pub struct Poll {
-    registry: Registry,
+    registry: Arc<Registry>,
```

Comment:
You will probably want to move this `Arc` into the wasi implementation of `Selector`.

Reply:
What would that look like? I just need a `Registry` object, not a reference, and not an object that gets "moved".